x-pack/test/upgrade/reporting_services.ts

```diff
@@ -0,0 +1,213 @@
+        })
+        .expect(200);
+
+      return async () => {
+        await esSupertest
+          .post('/_aliases')
+          .send({
+            actions: [
+              {
+                remove: { index: indexName, alias: `.reporting-${timestampForIndex}` },
+              },
+            ],
+          })
+          .expect(200);
+      };
+    },
+
+    async deleteAllReports() {
+      log.debug('ReportingAPI.deleteAllReports');
+
+      // ignores 409 errs and keeps retrying
+      await retry.tryForTime(5000, async () => {
+        await esSupertest
+          .post('/.reporting*/_delete_by_query')
+          .send({ query: { match_all: {} } })
+          .expect(200);
+      });
+    },
+
+    expectRecentPdfAppStats(stats: UsageStats, app: string, count: number) {
+      expect(stats.reporting.last_7_days.printable_pdf.app[app]).to.be(count);
+    },
+
+    expectAllTimePdfAppStats(stats: UsageStats, app: string, count: number) {
+      expect(stats.reporting.printable_pdf.app[app]).to.be(count);
+    },
+
+    expectRecentPdfLayoutStats(stats: UsageStats, layout: string, count: number) {
+      expect(stats.reporting.last_7_days.printable_pdf.layout[layout]).to.be(count);
+    },
+
+    expectAllTimePdfLayoutStats(stats: UsageStats, layout: string, count: number) {
+      expect(stats.reporting.printable_pdf.layout[layout]).to.be(count);
+    },
+
+    expectRecentJobTypeTotalStats(stats: UsageStats, jobType: string, count: number) {
+      expect(stats.reporting.last_7_days[jobType].total).to.be(count);
+    },
+
+    expectAllTimeJobTypeTotalStats(stats: UsageStats, jobType: string, count: number) {
+      expect(stats.reporting[jobType].total).to.be(count);
+    },
+
+    getCompletedReportCount(stats: UsageStats) {
+      return stats.reporting.status.completed;
+    },
+
+    expectCompletedReportCount(stats: UsageStats, count: number) {
+      expect(this.getCompletedReportCount(stats)).to.be(count);
+    },
+
+    getRecentPdfAppStats(stats: UsageStats, app: string) {
+      return stats.reporting.last_7_days.printable_pdf.app[app];
+    },
+
+    getAllTimePdfAppStats(stats: UsageStats, app: string) {
+      return stats.reporting.printable_pdf.app[app];
+    },
+
+    getRecentPdfLayoutStats(stats: UsageStats, layout: string) {
+      return stats.reporting.last_7_days.printable_pdf.layout[layout];
+    },
+
+    getAllTimePdfLayoutStats(stats: UsageStats, layout: string) {
+      return stats.reporting.printable_pdf.layout[layout];
+    },
+
+    getRecentJobTypeTotalStats(stats: UsageStats, jobType: string) {
+      return stats.reporting.last_7_days[jobType].total;
+    },
+
+    getAllTimeJobTypeTotalStats(stats: UsageStats, jobType: string) {
+      return stats.reporting[jobType].total;
+    },
+  };
+}
+
+export const services = {
+  ...xpackServices,
+  supertestWithoutAuth: apiIntegrationServices.supertestWithoutAuth,
+  usageAPI: apiIntegrationServices.usageAPI,
+  reportingAPI: ReportingAPIProvider,
+};
```

Comment:
I really like reporting API service and wonder if we can re-use for other functional/integration tests. If so, it could be in placed in api integration path, right?

Then upgrade_services may look like:

```
export const services = {
  ...xpackServices,
  supertestWithoutAuth: apiIntegrationServices.supertestWithoutAuth,
  usageAPI: apiIntegrationServices.usageAPI,
  reportingAPI: apiIntegrationServices.reportingAPI,
};
```

Just thoughts, I might not get the full context.

Reply:
Thanks @dmlemeshko sounds like a good idea, since it will impact both reporting and upgrade suites, let's do another PR to make a common service as an enhancement.  Will that work?